Sources/ConnectionInfo.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  ConnectionInfo.swift
+//  C7
+//
+//  Created by Scott Byrns on 4/1/16.
+//
+//
+public protocol ConnectionInfo {
+    var host: String? { get }
+    var port: Int? { get }
+    var connectionName: String? { get }
+    var username: String? { get }
+    var password: String? { get }
+}
```

Comment:
most of these could be replace by a single property of type `URI`. `URI` already holds `host`, `port`, `username` and `password`. 

`connectionName` should be renamed to `name` since we're talking about a connection the `connection` prefix is noise. Also why would this be needed?